lib/src/migrators/module/references.dart

```diff
@@ -84,57 +76,38 @@ class References {
   }
 
   /// Finds the original declaration of the variable referenced in [reference].
+  ///
+  /// This always returns [VariableDeclaration] or an [Argument], or null if the
+  /// declaration cannot be found.
   SassNode /*VariableDeclaration|Argument*/ originalDeclaration(
       VariableExpression reference) {
     var declaration = variables[reference];
     return variableReassignments[declaration] ?? declaration;
   }
 
-  /// For a given declaration or reference, returns the set of all nodes
-  /// associated with it, including itself.
-  ///
-  /// For mixins, this is the declaring MixinRule and all referencing
-  /// IncludeRules.
-  ///
-  /// For functions, this is the declaring FunctionRule and all referencing
-  /// FunctionRules (both regular and statically-determined get-function calls).
-  ///
-  /// For variables, this is the original declaration, all reassignments, and
-  /// all referencing VariableExpressions.
-  Set<SassNode> allNodes(SassNode node) {
-    if (node is IncludeRule) return allNodes(mixins[node]);
-    if (node is FunctionExpression) return allNodes(functions[node]);
-    if (node is VariableExpression) return allNodes(originalDeclaration(node));
-    if (node is MixinRule) return {node, ...mixins.keysForValue(node)};
-    if (node is FunctionRule) {
-      return {
-        node,
-        ...functions.keysForValue(node),
-        ...getFunctionReferences.keysForValue(node)
-      };
-    }
-    return {
-      node,
-      ...variables.keysForValue(node),
-      for (var reassignment in variableReassignments.keysForValue(node))
-        ...variables.keysForValue(reassignment)
-    };
-  }
+  References._(this.variables, this.variableReassignments, this.mixins,
+      this.functions, this.getFunctionReferences);
```

Comment:
Style nit: I'd do the unmodifiable wrapping here, rather than in `_ReferenceVisitor`, just because if this *were* a public constructor that's where it would be and it's nice to have that consistency.

Reply:
Done